webapi/src/main/scala/org/knora/webapi/routing/v1/ResourcesRouteV1.scala

```diff
@@ -289,52 +262,107 @@ object ResourcesRouteV1 extends Authenticator {
                 // Create a new resource with the given type, properties, and binary data (file) (non GUI-case).
                 // The binary data are contained in the request and have to be temporarily stored by Knora.
                 // For further details, please read the docs: Sipi -> Interaction Between Sipi and Knora.
-                entity(as[MultipartFormData]) { data => requestContext =>
+                entity(as[Multipart.FormData]) { formdata: Multipart.FormData => requestContext =>
 
+                    //println("/v1/resources - Multipart.FormData")
                     val requestMessageTry = Try {
 
+                        val userProfile = getUserProfileV1(requestContext)
+
+                        type Name = String
+
+                        @volatile var receivedFile: Option[File] = None
```

Comment:
Don't really know for sure, but this explanation seems plausible:

“… the volatile modifier guarantees that any thread that reads a field will see the most recently written value.”